src/backend/OpenCL/CHIPBackendOpenCL.cc

```diff
@@ -1260,46 +1320,430 @@ CHIPQueueOpenCL::enqueueBarrierImpl(std::vector<CHIPEvent *> *EventsToWaitFor) {
+    return false;
+
+  void *Dst = nullptr;
+  void *Src = nullptr;
+  const void *Symbol;
+  size_t SizeBytes;
+  size_t Offset;
+  hipMemcpyKind Kind;
+  Node->getParams(Dst, Symbol, SizeBytes, Offset, Kind);
+
+  hipError_t Err = hipGetSymbolAddress(&Src, Symbol);
+  if (Err != HIP_SUCCESS)
+    return false;
+
+  int Status = Exts->clCommandSVMMemcpyPOCL(
+      Handle, nullptr, Dst, (const char *)Src + Offset, SizeBytes,
+      SyncPointDeps.size(), SyncPointDeps.data(), SyncPoint, nullptr);
+
+  return Status == CL_SUCCESS;
+}
+
+// DONE
+bool CHIPGraphNativeOpenCL::addMemcpyNode(
+    CHIPGraphNodeMemcpyToSymbol *Node,
+    std::vector<cl_sync_point_khr> &SyncPointDeps,
+    cl_sync_point_khr *SyncPoint) {
+  if (!Exts->clCommandSVMMemcpyPOCL)
+    return false;
+
+  void *Dst = nullptr;
+  void *Src = nullptr;
+  const void *Symbol;
+  size_t SizeBytes;
+  size_t Offset;
+  hipMemcpyKind Kind;
+  Node->getParams(Src, Symbol, SizeBytes, Offset, Kind);
+
+  hipError_t Err = hipGetSymbolAddress(&Dst, Symbol);
+  if (Err != HIP_SUCCESS)
+    return false;
+
+  int Status = Exts->clCommandSVMMemcpyPOCL(
+      Handle, nullptr, (char *)Dst + Offset, Src, SizeBytes,
+      SyncPointDeps.size(), SyncPointDeps.data(), SyncPoint, nullptr);
+  return Status == CL_SUCCESS;
+}
+
+// DONE
+bool CHIPGraphNativeOpenCL::addMemsetNode(
+    CHIPGraphNodeMemset *Node, std::vector<cl_sync_point_khr> &SyncPointDeps,
+    cl_sync_point_khr *SyncPoint) {
+  if (!Exts->clCommandSVMMemfillRectPOCL)
+    return false;
+
+  hipMemsetParams Params = Node->getParams();
+
+  int Status;
+  size_t Region[3] = {Params.width, Params.height, 1};
+  Status = Exts->clCommandSVMMemfillRectPOCL(
+      Handle, nullptr, Params.dst,
+      nullptr,      // origin
+      Region,       // region
+      Params.pitch, // row pitch
+      0,            // slice pitch
+      (const void *)&Params.value, Params.elementSize, SyncPointDeps.size(),
+      SyncPointDeps.data(), SyncPoint, nullptr);
+  return Status == CL_SUCCESS;
+}
+#endif
+
+#ifdef cl_pocl_command_buffer_host_exec
+
+// DONE
+bool CHIPGraphNativeOpenCL::addHostNode(
+    CHIPGraphNodeHost *Node, std::vector<cl_sync_point_khr> &SyncPointDeps,
+    cl_sync_point_khr *SyncPoint) {
+  if (!Exts->clCommandHostFuncPOCL)
+    return false;
+
+  hipHostNodeParams Params = Node->getParams();
+
+  int Status;
+  Status = Exts->clCommandHostFuncPOCL(
+      Handle, nullptr, Params.fn, Params.userData, SyncPointDeps.size(),
+      SyncPointDeps.data(), SyncPoint, nullptr);
+  return Status == CL_SUCCESS;
+}
+
+// TODO output cl_event
+bool CHIPGraphNativeOpenCL::addEventRecordNode(
+    CHIPGraphNodeEventRecord *Node,
+    std::vector<cl_sync_point_khr> &SyncPointDeps,
+    cl_sync_point_khr *SyncPoint) {
+  if (!Exts->clCommandSignalEventPOCL)
+    return false;
+
+  CHIPEvent *E = Node->getEvent();
+  CHIPEventOpenCL *CLE = static_cast<CHIPEventOpenCL *>(E);
+
+  int Status;
+  // TODO BROKEN
```

Comment:
Perhaps clarify? (Is the driver broken or is the design broken?)

Reply:
fixed